ext/neural_snow/NeuralSnow.jl

```diff
@@ -0,0 +1,223 @@
+module NeuralSnow
+
+using Flux
+using ClimaLand
+using ClimaLand.Snow: AbstractDensityModel, SnowModel, SnowParameters
+import ClimaLand.Snow:
+    density_prog_vars,
+    density_prog_types,
+    density_prog_names,
+    update_density!,
+    update_density_prog!
+import ClimaLand.Parameters as LP
+using Thermodynamics
+
+using DataFrames, Dates, CSV, HTTP, Flux, cuDNN, StatsBase, BSON
+ModelTools = Base.get_extension(ClimaLand, :NeuralSnowExt).ModelTools
+#^^is this correct format since this is within the extension itself, or do i just use include(./ModelTools); using ModelTools; here?
+
+export NeuralDepthModel, snow_depth, update_density_prog!
+
+"""
+    NeuralDepthModel{FT <: AbstractFloat} <: AbstractDensityModel{FT}
+Establishes the density parameterization where snow density
+is calculated from a neural network determining the rate of change of snow depth, dzdt.
+"""
+struct NeuralDepthModel{FT} <: AbstractDensityModel{FT}
+    z_model::Flux.Chain
+    α::FT
+end
+
+"""
+    get_znetwork(; download_link = "https://caltech.box.com/shared/static/ay7cv0rhuiytrqbongpeq2y7m3cimhm4.bson")
+A default function for returning the snow-depth neural network from the paper.
+"""
+#This is the only place that creates the necessity of BSON for the neural extension and the make_model() from ModelTools. Keep or get rid of?
+#Can I change this to something CliMA already includes? is BSON vs JLD2 or anything else better? An artifact (can they store BSON/.jld2 files for direct usage without those packages?)?
```

Comment:
I would look into what Flux recommends for storing models. In the past I have had trouble reading (old) Flux models saved to BSON if Flux has updated.